cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -903,7 +903,8 @@ struct FileWriterHelper {
 
     ARROW_ASSIGN_OR_RAISE(buffer_, AllocateResizableBuffer(0));
     sink_.reset(new io::BufferOutputStream(buffer_));
-    ARROW_ASSIGN_OR_RAISE(writer_, NewFileWriter(sink_.get(), schema, options, metadata));
+    ARROW_ASSIGN_OR_RAISE(writer_,
+                          MakeFileWriter(sink_.get(), schema, options, metadata));
```

Comment:
```suggestion
                          MakeFileWriter(sink_, schema, options, metadata));
```

Reply:
`sink_` is a `unique_ptr` here, so not possible.